rts/Game/UnsyncedGameCommands.cpp

```diff
@@ -2726,14 +2726,16 @@ class NoLuaDrawActionExecutor : public IUnsyncedActionExecutor {
 class LuaUIActionExecutor : public IUnsyncedActionExecutor {
 public:
 	LuaUIActionExecutor() : IUnsyncedActionExecutor("LuaUI",
-			"Allows one to reload or disable LuaUI, or alternatively to send"
+			"Allows one to enable, reload or disable LuaUI,"
+			" execute an arbitrary Lua code, or alternatively to send"
```

Comment:
It would be good if it was possible to block this. I don't necessarily want people to execute arbitrary Lua code.

Reply:
It's modeled after BAR's `/execute` actually: https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/luaui/Widgets/api_unit_tracker_gl4.lua#L581-L604

Reply:
Yes but notice the above runs in the space of a specific widget. Widget handler (or the widget itself) can make sure that naughty API is not accessible. AFAICT the native `/execute` runs at the global level.

Reply:
Hmm I think overrides from WH should work the same way if called from Widget or globally. What kind of naughty API are you suspecting to be abused? If it's that bad I can move to LuaRules and require enabled cheats, but my reasoning was that a user can't executing anything more than they can execute from a widget.

Reply:
You could for example do `WG.referenceToANormallyBlockedFunction = Spring.GetVisibleProjectiles` to obtain a reference to a function which is normally blocked in widgets https://github.com/ZeroK-RTS/Zero-K/blob/master/LuaUI/cawidgets.lua#L597-L614

I think it would be best to add a modrule `allowUserLuaExecute = true|false` or something like that as a sort of safety valve (ZK would probably initially disable it because I'm slightly worried but at some point I would investigate and maybe enable).

Reply:
The json lib in use by bar has been recently updated and fixed by lazau (bluejay). Could it be put into engine?

Reply:
Sure, where is the relevant code?

Reply:
> Sure, where is the relevant code?

https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/common/luaUtilities/json.lua

Reply:
In Lua? Hmm that doesn't sound performant. Why not c++, given how many fast json libs are out there?

Reply:
I think Behe just misunderstood your "laundering results through JSON is **inelegant**" as "is **slow**". Choice of engine JSON lib seems irrelevant here since it's just for printing a value or two.

If by "putting the specific JSON lib used by BAR into engine" he meant putting it (or in general some Lua JSON lib) into basecontent then the answer to that is that it would be better to expose `Spring.{En,De}codeJSON` as normal engine interfaces as opposed to basecontent gadgets, see #877.

Reply:
Yes, can probably do https://github.com/beyond-all-reason/spring/issues/877